packages/api/src/routes/profileRoutes.js

```diff
@@ -14,29 +14,37 @@ profileRoutes.get('/user/:id', isAuthenticated, extractUser, async (req, res) =>
   const {
     email, firstName, lastName, profilePicUrl, batchId, role,
   } = user;
-  if (id.toLowerCase() === 'me') {
-    const projection = {
-      city: 1,
-      batchNumber: 1,
-    };
-    const batchOfUser = await Batch.findOne({ batchId: user.batchId }, projection);
-    const { city, batchNumber } = batchOfUser;
-    const finalUserDetails = {
-      email,
-      firstName,
-      lastName,
-      profilePicUrl,
-      batchId,
-      role,
-      city,
-      batchNumber,
-    };
-    res.json(finalUserDetails);
+  if (id.toLowerCase() !== 'me') {
+    return res.status(400).end();
   }
-  res.status(400).end();
+  if (role === 'instructor') {
+    return res.json(user);
+  }
+  // id = me and role = student
+  const projection = {
+    city: 1,
+    batchNumber: 1,
+  };
+  const batchOfUser = await Batch.findOne({ batchId: user.batchId }, projection);
+  const { city, batchNumber } = batchOfUser;
```

Comment:
const finalUserDetails = {...user, ...batchOfUser};

This will give us the user's _id, sex, and dob as well